leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/CaliforniumLwM2mRequestSender.java

```diff
@@ -47,6 +51,7 @@
     private final Set<Endpoint> endpoints;
     private final ObservationRegistry observationRegistry;
     private final LwM2mModelProvider modelProvider;
+    private final Map<String/* registration id */, ArrayList<Request> /* pending coap Request */> pendingRequests = new HashMap<>();
```

Comment:
Sure, the `ClientRegistry` should not handle pending request, but it's the central map for `Client`, regardless of its implementation.
The `CaliforniumLwM2mRequestSender.send` therefore has already a `Client` as parameter, so in my idea there is not to use a "second map/lookup" and instead use that Client. If that `Client` is extended by a List of `RequestCanceler` (see PR #102) it would be easy to cancel that list. But it was just an idea. Though this PR ist a Bugfix, I think it's better to find a solution then to discuss possibilities over and over. Therefore my proposal (see PR #103) is much closer to this solution than my "idea".